connexion/decorators/produces.py

```diff
@@ -57,17 +57,18 @@ def get_data_status_code(data):
         """
         url = flask.request.url
         logger.debug('Getting data and status code', extra={'data': data, 'data_type': type(data), 'url': url})
+        status_code, headers = 200, []
```

Comment:
Hi,

That's fine by me. My reasoning for using a list of tuples was in case the
sequencing of the headers was important to a user for some reason, but I
agree a dictionary is a simpler approach. Could we use an ordered
dictionary maybe?

I just had a quick look at the HTTP spec and it seems the ordering of
headers is not significant, except in the slightly more edge case of where
multiple headers with the same name are present:

"The order in which header fields with differing field names are received
is not significant. However, it is "good practice" to send general-header
fields first, followed by request-header or response- header fields, and
ending with the entity-header fields.

Multiple message-header fields with the same field-name MAY be present in a
message if and only if the entire field-value for that header field is
defined as a comma-separated list [i.e., #(values)]. It MUST be possible to
combine the multiple header fields into one "field-name: field-value" pair,
without changing the semantics of the message, by appending each subsequent
field-value to the first, each separated by a comma. The order in which
header fields with the same field-name are received is therefore
significant to the interpretation of the combined field value, and thus a
proxy MUST NOT change the order of these field values when a message is
forwarded."

( http://tools.ietf.org/html/rfc2616#page-31 )

It's your call, what shall we go with, list of tuples, ordered dictionary
or plain dictionary?

Dan

On Wed, 28 Oct 2015 07:16 João Santos <notifications@github.com> wrote:

> In connexion/decorators/produces.py
> <https://github.com/zalando/connexion/pull/77#discussion_r43223588>:
>
> > @@ -57,17 +57,18 @@ def get_data_status_code(data):
> >          """
> >          url = flask.request.url
> >          logger.debug('Getting data and status code', extra={'data': data, 'data_type': type(data), 'url': url})
> > +        status_code, headers = 200, []
>
> I would make headers None or {} by default.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zalando/connexion/pull/77/files#r43223588>.
>